src/Api/JsonRpc/Client.php

```diff
@@ -428,6 +428,6 @@ private function getCaseRunStatusIdByResult(string $result): int
             return (int) $this->testExecutionStatus[$result]['id'];
         }
 
-        return $this->testExecutionStatus['error']['id'];
+        return $this->testExecutionStatus['failed']['id'];
```

Comment:
There may not be a status with name `failed`. You have to rework this logic to do the following: 

1) Search by the preferred name. If not found
2) Search by weight. Errors and Failures are with negative weight - take the first one (or the last one, doesn't matter, just be consistent).  If not found (should not happen b/c BE is trying to prevent that)
3) Raise exception & crash.